std/algorithm/comparison.d

```diff
@@ -736,6 +736,28 @@ template equal(alias pred = "a == b")
         {
             return r1 == r2;
         }
+        // if one of the arguments is a string and the other isn't, then auto-decoding
+        // can be avoided if they have the same ElementEncodingType
+        else static if (is(typeof(pred) == string) && pred == "a == b" &&
+            anySatisfy!(isSomeString, Range1, Range2) &&
+            anySatisfy!(templateNot!isSomeString, Range1, Range2) &&
```

Comment:
Does this mean "exactly one of `Range1` and `Range2` is a string?" In that case: `isSomeString!Range1 != isSomeString!Range2`